client/src/components/statCalculations/AssignDisplay.js

```diff
@@ -0,0 +1,96 @@
+import styled from "styled-components";
+import React, { Component, createRef } from "react";
+
+const Wrapper = styled.div`
+  width: 100%;
+  height: 15vh;
+  display: flex;
+  justify-content: center;
+`;
+export default class AssignDisplay extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // statIndex tracks the index of what stat we are setting, its always one ahead
+      statIndex: 0,
+      buttonsDisabled: [false, false, false, false, false, false]
+    };
+  }
+
+  createButtons = statRolls => {
+    // temporary button array to return, one for each stat and a clear
+    let tmpButArr = [];
+
+    for (let i = 0; i < statRolls.length; i++) {
+      tmpButArr.push(
+        <button
+          key={i}
+          type="button"
+          disabled={this.state.buttonsDisabled[i]}
+          onClick={() => {
+            this.props.assignStat(statRolls[i], this.state.statIndex);
+            // I want to update only an index of the array in state
+            // So I'm making a copy, changing it, and setting state variable to it
+            let tempStateCopy = this.state.buttonsDisabled;
+            tempStateCopy[i] = true;
+
+            if (this.state.statIndex < statRolls.length - 1) {
+              this.setState({
+                buttonsDisabled: tempStateCopy,
+                statIndex: (this.state.statIndex += 1)
+              });
+            } else {
+              this.setState({
+                buttonsDisabled: tempStateCopy
+              });
+            }
+          }}
+        >
+          {statRolls[i]}
+        </button>
+      );
+    }
+
+    // add a clear button at the end to reset stat allocation
+    tmpButArr.push(
+      <button
+        key={statRolls.length + 1}
+        type="button"
+        onClick={() => {
```

Comment:
Line 59: should be it's own function. Really hard to read here. () => this.handleEtcEtc()

Reply:
done, handleClear()